CodeEdit/Features/CEWorkspace/Models/Resource.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  CEWorkspaceActor.swift
+//  CodeEdit
+//
+//  Created by Wouter Hennen on 19/06/2023.
+//
+
+import Foundation
+import AppKit
+import SwiftUI
+
+protocol Resource: AnyObject, Identifiable {
+    var name: String { get set }
+    var url: URL { get set }
+
+    var parentFolder: Folder? { get set }
+
+    func resolveItem(components: [String]) -> any Resource
+
+    func update(with url: URL) throws
+
+    var iconColor: Color { get }
+
+    var systemImage: String { get }
+
+    func fileName(typeHidden: Bool) -> String
+}
+
+extension Resource {
```

Comment:
Why are these properties in an extension?

Reply:
Protocols cannot have function (or property) implementations, only declarations. To add default implementations you need to use an extension